File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(firstStrand, secondStrand) {
  var distance = 0;

  if (firstStrand.length !== secondStrand.length) {
    throw('DNA strands must be of equal length.');
  }

  for (var i = 0; i <= firstStrand.length; i++) {
    if (firstStrand.charAt(i) !== secondStrand.charAt(i)) {
      distance++;
    }
  }

  return distance;
}

module.exports = Hamming;

File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand_one, strand_two)
    distance = 0
    strand_one.split.each.with_index do |nucleotide, i|
      if nucleotide == strand_two[i]
        distance += 1
      end
    end
    distance
  end
end
